tmt/ansible.py

```diff
@@ -0,0 +1,308 @@
+"""
+Ansible integration for tmt.
+
+This module provides classes and utilities for managing Ansible inventory generation
+and configuration within tmt test plans.
+"""
+
+from typing import TYPE_CHECKING, Any, Optional, Union, cast
+
+from typing_extensions import TypedDict
+
+import tmt.log
+import tmt.utils
+from tmt._compat.pathlib import Path
+from tmt.container import SerializableContainer, container, field
+
+if TYPE_CHECKING:
+    from tmt.steps.provision import Guest
+
+
+class _RawGuestAnsible(TypedDict, total=False):
+    """Raw input data for GuestAnsible.from_spec()"""
+
+    group: Optional[str]
+    vars: Optional[dict[str, Any]]
+
+
+class _RawPlanAnsibleInventory(TypedDict, total=False):
+    """Raw input data for PlanAnsibleInventory.from_spec()"""
+
+    layout: Optional[str]
+
+
+def normalize_plan_ansible(
+    key_address: str,
+    raw_ansible: Any,
+    logger: tmt.log.Logger,
+) -> 'PlanAnsible':
+    """
+    Normalize a ``ansible`` key value.
+
+    :param key_address: location of the key being that's being normalized.
+    :param logger: logger to use for logging.
+    :param raw_ansible: input from either command line or fmf node.
+    """
+    return PlanAnsible.from_spec(raw_ansible)
+
+
+def normalize_guest_ansible(
+    key_address: str,
+    raw_ansible: Any,
+    logger: tmt.log.Logger,
+) -> 'GuestAnsible':
+    """
+    Normalize a ``ansible`` key value from provision guest data.
+
+    :param key_address: location of the key being that's being normalized.
+    :param logger: logger to use for logging.
+    :param raw_ansible: input from either command line or fmf node.
+    """
+    return GuestAnsible.from_spec(raw_ansible)
+
+
+@container
+class GuestAnsible(SerializableContainer):
+    """
+    Ansible configuration for individual guests.
+    """
+
+    group: Optional[str] = field(
+        default=None,
+        help='Assigns the guest to a specific Ansible group.',
+    )
+    # TODO Add typing for vars raw input data
+    vars: dict[str, Any] = field(  # pyright: ignore[reportUnknownVariableType]
+        default_factory=dict,
+        help=(
+            'Defines host-specific Ansible variables to include under that host in the inventory.'
+        ),
+    )
+
+    @classmethod
+    def from_spec(cls, spec: Union[_RawGuestAnsible, None]) -> 'GuestAnsible':
```

Comment:
```suggestion
    def from_spec(cls, spec: Optional[_RawGuestAnsible]) -> 'GuestAnsible':
```
Equivalent, but we use this convention. Similar below